File: jsfs-ser/cs/src-ser/com/wilutions/jsfs/FileSystemNotify.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using byps;

namespace com.wilutions.jsfs
{
	
	/// <summary>
	/// Notification interface.
	/// </summary>
	/// <remarks>
	/// This interface is used by the JFSF Agent to notify the browser about an event.
	/// </remarks>
	// checkpoint byps.gen.cs.GenRemoteClass:93
	public interface FileSystemNotify : BRemote {
	
		/// <summary>
		/// Notify the browser about an event.
		/// </summary>
		void Notify(NotifyInfo notifyInfo);
		void Notify(NotifyInfo notifyInfo, BAsyncResult<Object> asyncResult) ;
		/// <summary>
		/// Notify the browser about an event.
		/// </summary>
		Task NotifyAsync(NotifyInfo notifyInfo);
		
		
	}
}  // end namespace

File: jsfs-api/src/com/wilutions/jsfs/FileSystemNotify.java
/* USE THIS FILE ACCORDING TO THE COPYRIGHT RULES IN LICENSE.TXT WHICH IS PART OF THE SOURCE CODE PACKAGE */ 
package com.wilutions.jsfs;

import byps.BRemote;
import byps.RemoteException;

/**
 * Notification interface.
 * This interface is used by the JFSF Agent to notify the browser about an event.
 * @BClientRemote
 */
public interface FileSystemNotify extends BRemote {

  /**
   * Notify the browser about an event.
   * @param notifyInfo NotifyInfo or ExecuteNotifyInfo object.
   * @throws RemoteException
   */
  public void notify(NotifyInfo notifyInfo) throws RemoteException;
  
}
